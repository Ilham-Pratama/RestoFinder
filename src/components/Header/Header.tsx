import ThemedText from 'components/ThemedText';
import React from 'react';
import { StyleSheet } from 'react-native';
import sharedStyles from 'shared/styles';

interface HeaderProps {
  children: any;
}

const Header = ({ children }: HeaderProps) => {
  return (
    <ThemedText
      accessibilityRole="header"
      style={[
        styles.header,
        sharedStyles.marginBottom2,
        sharedStyles.marginHorizontal3
      ]}>
      {children}
    </ThemedText>
  );
};

const styles = StyleSheet.create({
  header: {
    fontSize: 22,
    fontWeight: '700'
  }
});

export default Header;
