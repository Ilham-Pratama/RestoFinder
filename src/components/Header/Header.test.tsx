import React from 'react';
import { render } from '@testing-library/react-native';
import Header from '.';

describe('Header', () => {
  it('renders Header', () => {
    const children = 'Header Example';
    const component = render(<Header>{children}</Header>);

    const text = component.getByRole('header');

    expect(component.toJSON()).toMatchSnapshot();
    expect(text.props.style[0].length).toBe(3);
    expect(text.props.children).toBe(children);
  });
});
