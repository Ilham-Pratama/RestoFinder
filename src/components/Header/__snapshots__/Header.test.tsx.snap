// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header 1`] = `
<Text
  accessibilityRole="header"
  style={
    Array [
      Array [
        Object {
          "fontSize": 22,
          "fontWeight": "700",
        },
        Object {
          "marginBottom": 10,
        },
        Object {
          "marginHorizontal": 15,
        },
      ],
      Object {
        "color": "rgb(28, 28, 30)",
      },
    ]
  }
>
  Header Example
</Text>
`;
